automerge-wasm/README.md

```diff
@@ -1,696 +1,478 @@
+
+The `merge()` command applies all changes in the argument doc into the calling doc.  Therefore if doc a has 1000 changes that doc b lacks and doc b has only 10 changes that doc a lacks, `a.merge(b)` will be much faster than `b.merge(a)`.
+
+```javascript
+    let doc1 = create()
+    doc1.put("_root", "key1", "val1")
+
+    let doc2 = doc1.fork()
+
+    doc1.put("_root", "key2", "val2")
+    doc2.put("_root", "key3", "val3")
+
+    doc1.merge(doc2)
+
+    doc1.materialize("_root")       // returns { key1: "val1", key2: "val2", key3: "val3" }
+    doc2.materialize("_root")       // returns { key1: "val1", key3: "val3" }
+
+    doc1.free(); doc2.free()
+```
+
+Note that calling `a.merge(a)` will produce an unrecoverable error from the wasm-bindgen layer which (as of this writing) there is no workaround for.
+
+### Saving and Loading
+
+Calling `save()` converts the document to a compressed `Uint8Array()` that can be saved to durable storage.  This format uses a columnar storage format that compresses away most of the Automerge metadata needed to manage the CRDT state, but does include all of the change history.
+
+If you wish to incrementally update a saved Automerge doc you can call `saveIncremental()` to get a `Uint8Array()` of bytes that can be appended to the file with all the new changes(). Note that the `saveIncremental()` bytes are not as compressed as the whole document save as each chunk has metadata information needed to parse it.  It may make sense to periodically perform a new `save()` to get the smallest possible file footprint.
+
+The `load()` function takes a `Uint8Array()` of bytes produced in this way and constitutes a new document.  The `loadIncremental()` method is available if you wish to consume the result of a `saveIncremental()` with an already instanciated document.
+
+```javascript
+  import { create, load } from "automerge-wasm"
+
+  let doc1 = create()
+
+  doc1.put("_root", "key1", "value1")
+
+  let save1 = doc1.save()
+
+  let doc2 = load(save1)
+
+  doc2.materialize("_root")  // returns { key1: "value1" }
+
+  doc1.put("_root", "key2", "value2")
+
+  let saveIncremental = doc1.saveIncremental()
+
+  let save2 = doc1.save()
+
+  let save3 = new Uint8Array([... save1, ... saveIncremental])
+
+  // save2 has fewer bytes than save3 but contains the same ops
+
+  doc2.loadIncremental(saveIncremental)
+
+  let doc3 = load(save2)
+
+  let doc4 = load(save3)
+
+  doc1.materialize("_root")  // returns { key1: "value1", key2: "value2" }
+  doc2.materialize("_root")  // returns { key1: "value1", key2: "value2" }
+  doc3.materialize("_root")  // returns { key1: "value1", key2: "value2" }
+  doc4.materialize("_root")  // returns { key1: "value1", key2: "value2" }
+
+  doc1.free(); doc2.free(); doc3.free(); doc4.free()
+```
+
+### Syncing
```

Comment:
I think this could do with a small narrative section about the sync protocol needing a number of round trips. Especially the fact that `generateSyncMessage` returns `null` when sync is complete.